shared/src/main/scala/mlscript/codegen/typescript/TsTypegen.scala

```diff
@@ -83,19 +84,18 @@ final class TsTypegenCodeBuilder {
     tySymb match {
       case (classInfo: ClassSymbol) => addTypeGenClassDef(classInfo, methods)
       case (aliasInfo: TypeAliasSymbol) => addTypeGenTypeAlias(aliasInfo)
-      case (traitInfo: TraitSymbol) => throw CodeGenError("Typegen for traits is not supported currently")
+      case (traitInfo: TraitSymbol) => addTypegenTraitDef(traitInfo, methods)
     }
   }
 
   /**
     * Translate class method to typescript declaration
     *
-    * @param classInfo
+    * @param topTypeParams top level params from class or trait declaration
```

Comment:
What does "top"refer to in "topTypeParams"?

Reply:
class or trait type parameters, this is so that the same method can be used to declare methods from both classes and traits. For e.g. the `T` in `Box[T]`.

Reply:
My question still stands. Why "top"? Maybe naming it "ownerTypeParams" will be clearer?

Reply:
I'll remove the "top" thing.

```
     * @param declarationTypeParams type params from the class or trait declaring the method
```